framework/scripts/agent_upgrade.py

```diff
@@ -82,68 +79,83 @@ def main():
-            else:
-                print(upgrade_command_result)
-
-        counter = 0
-        agent_info_stat = os.stat(agent_info).st_mtime
-
-        sleep(10)
-        while agent_info_stat == os.stat(agent_info).st_mtime and counter < common.agent_info_retries:
-            sleep(common.agent_info_sleep)
-            counter = counter + 1
-
-        if agent_info_stat == os.stat(agent_info).st_mtime:
-            raise WazuhException(1716, "Timeout waiting for agent reconnection.")
-
-        upgrade_result = agent.upgrade_result(debug=args.debug)
-        if not args.silent:
-            print(upgrade_result)
-
+        upgrade_command_result = send_task_upgrade_module(command='upgrade', file_path=args.file,
+                                                            agent_list=agent_list,
+                                                            installer=args.execute if args.execute else "upgrade.sh",
+                                                            debug=args.debug,
+                                                            show_progress=print_progress if not args.silent else None,
+                                                            chunk_size=args.chunk_size,
+                                                            rl_timeout=-1 if args.timeout == None else args.timeout)
     # WPK upgrade file
     else:
-        prev_ver = agent.version
-        upgrade_command_result = agent.upgrade(wpk_repo=args.repository, debug=args.debug, version=args.version,
-                                               force=args.force,
-                                               show_progress=print_progress if not args.silent else None,
-                                               chunk_size=args.chunk_size,
-                                               rl_timeout=-1 if args.timeout == None else args.timeout, use_http=use_http)
+        #prev_ver = agent.version
+        upgrade_command_result = send_task_upgrade_module(command='upgrade', 
+                                                            agent_list=agent_list,
+                                                            wpk_repo=args.repository,
+                                                            debug=args.debug, 
+                                                            version=args.version,
+                                                            force_upgrade=args.force,
+                                                            show_progress=print_progress if not args.silent else None,
+                                                            chunk_size=args.chunk_size,
+                                                            rl_timeout=-1 if args.timeout == None else args.timeout, 
+                                                            use_http=use_http)
+
+    counter = 0
+    
+    sleep(10)
+
+    #check if any agent start the upgrade
+    agents = json.loads(upgrade_command_result.replace("\'", "\""))
```

Comment:
If there is an [exception](https://github.com/wazuh/wazuh/pull/5477/files#diff-f591bbdd4c283c1e11856644cad66bcbR1669) when sending the message to the upgrade socket, I think you should check that before this.

Reply:
you are right!